File: sdk-python/examples/smart_random.py
"""Smart random bot — picks a random non-wall direction each turn."""

import random

from pyrat_sdk import Bot, Context, Direction, GameState


class SmartRandom(Bot):
    name = "SmartRandom"
    author = "PyRat SDK"

    def think(self, state: GameState, ctx: Context) -> Direction:
        # get_effective_moves() returns Direction values for non-wall moves
        # from the current position (UP, RIGHT, DOWN, LEFT — never STAY).
        moves = state.get_effective_moves()
        if moves:
            return random.choice(moves)
        return Direction.STAY


if __name__ == "__main__":
    SmartRandom().run()

File: sdk/rust/examples/smart_random.rs
use pyrat_sdk::{Bot, Context, Direction, GameState, Options};
use rand::seq::IndexedRandom;

struct SmartRandom;

impl Options for SmartRandom {}

impl Bot for SmartRandom {
    fn think(&mut self, state: &GameState, _ctx: &Context) -> Direction {
        let moves = state.effective_moves(None);
        *moves.choose(&mut rand::rng()).unwrap_or(&Direction::Stay)
    }
}

fn main() {
    pyrat_sdk::run(SmartRandom, "Smart Random", "PyRat");
}
